### PR TITLE
mypy: improve config for tests

### PR DESCRIPTION
**Pull request by blueyed**

### CODE REVIEW

**Comment by bbayles** [comment]
Could you summarize why these changes are improvements?

**Comment by blueyed** [comment]
I'm not too convinced myself, but:

1) `[mypy-more_itertools.tests.*]` appears to be wrong, since tests are in `tests` directly.
2) `ignore_errors` seems to be not helpful when you want to do type checking explicitly.
3) `disallow_untyped_defs = false` is used since tests are not typed.

Altogether this allows for `mypy tests`, which still results in a lot of errors, but e.g. showed https://github.com/erikrose/more-itertools/pull/377.
